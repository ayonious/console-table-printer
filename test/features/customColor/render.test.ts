import { Table } from '../../../index';
import { getTableBody, getTableHeader } from '../../testUtils/getRawData';

describe('Rendered Table content verification', () => {
  it('should make sure each column is what its expected to be', () => {
    const p = new Table({
      shouldDisableColors: true,
      columns: [
        { name: 'red_left_align_index', alignment: 'left', color: 'red' },
        { name: 'right_align_text', alignment: 'right' },
        { name: 'green_value', color: 'green' },
      ],
      colorMap: {
        custom_green: '\x1b[32m', // define customized color
      },
    });

    p.addRows([
      {
        red_left_align_index: 2,
        right_align_text: 'This is my defined Green',
        green_value: 10.212,
      },
      {
        red_left_align_index: 3,
        right_align_text: 'This row is blue as well',
        green_value: 10.212,
      },
    ]);

    const [renderedHeader, renderedBody] = [getTableHeader(p), getTableBody(p)];
    expect(renderedHeader).toEqual(
      '│ red_left_align_index │         right_align_text │ green_value │'
    );
    expect(renderedBody).toEqual([
      '│ 2                    │ This is my defined Green │      10.212 │',
      '│ 3                    │ This row is blue as well │      10.212 │',
    ]);

    // Add snapshot test
    expect(p.render()).toMatchSnapshot();
  });
});
