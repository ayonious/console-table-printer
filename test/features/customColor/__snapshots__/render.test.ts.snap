// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendered Table content verification should make sure each column is what its expected to be 1`] = `
"┌──────────────────────┬──────────────────────────┬─────────────┐
│ red_left_align_index │         right_align_text │ green_value │
├──────────────────────┼──────────────────────────┼─────────────┤
│ 2                    │ This is my defined Green │      10.212 │
│ 3                    │ This row is blue as well │      10.212 │
└──────────────────────┴──────────────────────────┴─────────────┘"
`;
